app/javascript/components/DonateLink.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import CurrencyAmount from './CurrencyAmount';
+import ProcessingThen from './ProcessingThen';
+import { FormattedMessage } from 'react-intl';
+import classNames from 'classnames';
+import './DonateLink.css';
+
+export default props => {
+  let buttonId = 'fundraiser.donation_once';
+  let buttonText = 'Donate {amount} Just Once';
+  if (props.recurring || props.name == 'recurring') {
+    if (props.weekly) {
+      buttonId = 'fundraiser.donation_weekly';
+      buttonText = 'Donate {amount} Weekly';
+    } else {
+      buttonId = 'fundraiser.donation_recurring';
+      buttonText = 'Donate {amount} Monthly';
+    }
+  }
+
+  if (props.recurringDonor) {
+    buttonId = 'fundraiser.donate_amount';
+    buttonText = 'Donate {amount}';
+  }
+  const className = classNames(
+    'DonateLink',
+    props.disabled ? 'disabled' : '',
+    props.className ? props.className : ''
+  );
+  return (
+    <button
+      id={props.id}
+      className={className}
+      name={props.name}
+      disabled={props.disabled}
+      onClick={props.disabled ? null : props.onClick}
+    >
+      <ProcessingThen processing={props.submitting || false}>
+        {'Or'}
```

Comment:
```suggestion

```

Reply:
> @tahaf22labs You can simplify this; instead of creating a new component that is almost an exact copy of the previous one to make it look different, you can modify the existing one to match the requirements.
> 
> Since you are using a button and only modifying the styling, then you can reuse the `DonateButton` component and modify the `Button` component to now receive an extra property; take this as an example:
> 
> ```js
> const Button = ({ onClick, className, disabled, theme, type, name }) => {
>     const classes = classNames(styles.button, styles[theme], className);
>     // theme being the new property that will help you define what styles to use
>     // styles.button being the common styles between the two (if any)
> ```
> 
> Then when you're using it:
> 
> For the monthly Button:
> 
> ```
> <DonateButton
>         currency={currency}
>         amount={donationAmount || 0}
>         submitting={this.state.submitting}
>         name="recurring"
>         recurring={true}
>         recurringDonor={this.state.recurringDonor}
>         weekly={this.props.weekly}
>         disabled={this.disableSubmit()}
>         onClick={e => this.onClickHandle(e)}
>         theme="primary"
>  />
> ```
> 
> For the one-off link:
> 
> ```
> <DonateButton
>         currency={currency}
>         amount={donationAmount || 0}
>         submitting={this.state.submitting}
>         name="one_time"
>         recurring={false}
>         recurringDonor={this.state.recurringDonor}
>         disabled={this.disableSubmit()}
>         onClick={e => this.onClickHandle(e)}
>         theme="secondary"
>  />
> ```
> 
> And finally, on the Button component:
> 
> ```
> < Button
>       className="DonateButton"
>       name={props.name}
>       onClick={props.onClick}
>       disabled={props.disabled}
>       theme={props.theme}
>     >
> ```
> 
> Then on the `Button.scss` you will have two different themes (primary being the styles of the regular Button and secondary being the styles for the link).
> 
> Could you please add screenshots to the PR description? (covering how the change looks on desktop and mobile view).
> 
> Please let me know if you have any questions.

Thank you for your precise feedback! I've updated the CSS, Can you please have a look and let me know if that fits!